ddtrace/internal/coverage/instrumentation_py3_12.py

```diff
@@ -9,44 +9,129 @@
+                    if code.co_name == "<module>" and len(lines) == 1 and package is not None:
+                        import_names[line] = (package, ("",))
+
+                if opcode is EXTENDED_ARG:
+                    ext.append(arg)
+                    continue
+                else:
+                    _previous_previous_arg = previous_arg
+                    previous_arg = current_arg
+                    current_arg = int.from_bytes([*ext, arg], "big", signed=False)
+                    ext.clear()
+
+                if opcode == IMPORT_NAME:
+                    import_depth: int = code.co_consts[_previous_previous_arg]
+                    current_import_name: str = code.co_names[current_arg]
+                    # Adjust package name if the import is relative and a parent (ie: if depth is more than 1)
+                    current_import_package: str = (
+                        ".".join(package.split(".")[: -import_depth + 1]) if import_depth > 1 else package
+                    )
+
+                    if line in import_names:
+                        import_names[line] = (
+                            current_import_package,
+                            tuple(list(import_names[line][1]) + [current_import_name]),
+                        )
+                    else:
+                        import_names[line] = (current_import_package, (current_import_name,))
+
+                # Also track import from statements since it's possible that the "from" target is a module, eg:
+                # from my_package import my_module
+                # Since the package has not changed, we simply extend the previous import names with the new value
+                if opcode == IMPORT_FROM:
+                    import_from_name = f"{current_import_name}.{code.co_names[current_arg]}"
+                    if line in import_names:
+                        import_names[line] = (
+                            current_import_package,
+                            tuple(list(import_names[line][1]) + [import_from_name]),
+                        )
+                    else:
+                        import_names[line] = (current_import_package, (import_from_name,))
+
+        except StopIteration:
+            pass
 
         # Recursively instrument nested code objects
         for nested_code in (_ for _ in code.co_consts if isinstance(_, CodeType)):
-            _, nested_lines = instrument_all_lines(nested_code, hook, path)
+            _, nested_lines = instrument_all_lines(nested_code, hook, path, package)
             lines.update(nested_lines)
 
         # Register the hook and argument for the code object
-        _CODE_HOOKS[code] = (hook, path)
+        _CODE_HOOKS[code] = (hook, path, import_names)
+
+        # Special case for empty modules (eg: __init__.py ):
+        # Make sure line 0 is marked as executable, and add package dependency
+        if not lines and code.co_name == "<module>" and code.co_code == bytes([151, 0, 121, 0]):
```

Comment:
Can we replace the literal bytes with constants from `dis` to make this more readable?